.gitignore

```diff
@@ -1,9 +1,11 @@
 reference
-reference/
 releases
-releases/
 build
-build/
+demos/node/output
+
+.DS_Store
+node_modules
```

Comment:
@rvilarl Sorry for the confusion. I actually moved node_modules up nearer the top because it seemed important. The stuff at the bottom seems like it is either legacy, or copied from another .gitignore. Eventually we might consider deleting it (e.g., svn). I guess it's possible someone's using intellij to work on VexFlow.